### PR TITLE
Add more tests to ResultResource

### PR DESCRIPTION
**Pull request by nicolasruscher**
### Checklist
- [ ] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [X] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [X] Server: I added multiple integration tests (Spring) related to the features (with a high test coverage)

### Motivation and Context

We want to enhance the Test Coverage for the Server tests. In this PR we enhance the test coverage for the ResultResource to be > 80 % 

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration
3. ...

### Test Coverage

ResultResource: > 92%

### CODE REVIEW

File: src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java [hunk 1]
```diff
@@ -490,6 +505,44 @@ public void getResult_asStudent() throws Exception {
...
+        modelingSubmission.submitted(true);
+        modelingSubmission.submissionDate(now.minusHours(3));
+        participation.addSubmission(modelingSubmission);
+        modelingSubmission.setParticipation(participation);
+        submissionRepository.save(modelingSubmission);
+        studentParticipationRepository.save(participation);
+
+        List<Result> results = request.getList("/api/exercises/" + this.examModelingExercise.getId() + "/results", HttpStatus.OK, Result.class);
+        assertThat(results).hasSize(5);
+        // TODO: check additional values
```
Comment (FuchsDominik) [inline src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java:543]: Am I a bit premature or did you forget to remove this comment?

File: src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java [hunk 2]
```diff
@@ -490,6 +505,44 @@ public void getResult_asStudent() throws Exception {
...
+    @WithMockUser(value = "instructor1", roles = "INSTRUCTOR")
+    public void testGetResultsForExamExercise() throws Exception {
+        var now = ZonedDateTime.now();
+        for (int i = 1; i <= 10; i++) {
+            ModelingSubmission modelingSubmission = new ModelingSubmission();
+            modelingSubmission.model("Text");
+            modelingSubmission.submitted(true);
+            modelingSubmission.submissionDate(now.minusHours(3));
+            database.addSubmission(this.examModelingExercise, modelingSubmission, "student" + i);
+            if (i % 3 == 0) {
```
Comment (FuchsDominik) [inline src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java:518]: what's the idea behind only adding results to submission 3, 4, 6, 8, 9 ?

File: src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java (hunk 1 above)
Comment (nicolasruscher) [reply src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java:543]: yeah I will remove it :D

File: src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java (hunk 2 above)
Comment (nicolasruscher) [reply src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java:518]: not a specific one, just used to have different submissions

**Review by FrankeLukas — APPROVED** [review]
Changes look ok to me.

**Review by ivanchimeno — APPROVED** [review]
lgtm

File: src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java [hunk 3]
```diff
@@ -490,6 +503,38 @@ public void getResult_asStudent() throws Exception {
...
+        modelingSubmission.model("Text");
+        modelingSubmission.submitted(true);
+        modelingSubmission.submissionDate(now.minusHours(3));
+        participation.addSubmission(modelingSubmission);
+        modelingSubmission.setParticipation(participation);
+        submissionRepository.save(modelingSubmission);
+        studentParticipationRepository.save(participation);
+
+        List<Result> results = request.getList("/api/exercises/" + this.examModelingExercise.getId() + "/results", HttpStatus.OK, Result.class);
+        assertThat(results).hasSize(5);
```
Comment (MDK5) [inline src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java:535]: Correct me if I'm wrong, but I think the list could/should be verified by more parameters than just the size. Maybe store the entries added above and check if the server sends back those entries.

**Review by MDK5 — APPROVED** [review]
Looks good to me, just a short suggestion

**Review by krusche — APPROVED** [review]

**Pull request merged**